devito/finite_differences/differentiable.py

```diff
@@ -526,7 +536,7 @@ def __new__(cls, expr, dimensions, **kwargs):
                     raise ValueError("Dimension `%s` already appears in a "
                                      "nested tensor contraction" % d)
 
-        obj = sympy.Expr.__new__(cls, expr, *dimensions)
+        obj = sympy.Expr.__new__(cls, expr)
```

Comment:
Now it makes sense to me, while I find the way it was conceptually wrong.

If you pass them to the superclass constructor you're signaling SymPy that a Dimension is an argument, so it's part of the expression tree, it's traversable, it may have children args, etc etc. But it's not what a Dimension represents here. It's rather a property of the IndexDerivative, the same way the `name` is a property of a `Symbol` (not a (traversable) arg)

Reply:
ok